src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,44 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Machine;
+import core.mate.academy.service.impl.BulldozerProducer;
+import core.mate.academy.service.impl.ExcavatorProducer;
+import core.mate.academy.service.impl.TruckProducer;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    private final HashMap<String, MachineProducer<? extends Machine>> map;
```

Comment:
Better to use the class as a key 
```suggestion
    private final HashMap<Class<?>, MachineProducer<? extends Machine>> map;
```